byzcoin/bcadmin/test.sh

```diff
@@ -28,4 +33,80 @@ testCreateStoreRead(){
     testFail ./bcadmin add spawn:xxx
 }
 
+testAddDarc(){
+  runCoBG 1 2 3
+    runGrepSed "export BC=" "" ./"$APP" create --roster public.toml --interval .5s
+  eval $SED
+  [ -z "$BC" ] && exit 1
+
+  echo -e "\n\nTESTING ADDITION\n"
+  testOK ./"$APP" darc_add
+  testOK ./"$APP" darc_add -out_id ./darc_id.txt
+  testOK ./"$APP" darc_add
+  ID=`cat ./darc_id.txt`
+  echo -e "\n\nTESTING SHOW\n"
+  testOK ./"$APP" darc_show --id "$ID"
+}
+
+testRuleDarc(){
+  runCoBG 1 2 3
+    runGrepSed "export BC=" "" ./"$APP" create --roster public.toml --interval .5s
+  eval $SED
+  [ -z "$BC" ] && exit 1
+
+  testOK ./"$APP" darc_add -out_id ./darc_id.txt -out_key ./darc_key.txt
+  ID=`cat ./darc_id.txt`
+  KEY=`cat ./darc_key.txt`
+  testOK ./"$APP" darc_rule_add spawn:xxx -identity ed25519:foo -darc "$ID" -signer "$KEY"
+  testOK ./"$APP" darc_show -id "$ID"
+  testOK ./"$APP" darc_rule_update spawn:xxx -identity "ed25519:foo | ed25519:oof" -darc "$ID" -signer "$KEY"
+  testOK ./"$APP" darc_show -id "$ID"
+  testOK ./"$APP" darc_rule_del spawn:xxx -darc "$ID" -signer "$KEY"
+  testOK ./"$APP" darc_show -id "$ID"
+}
+
+testAddDarcFromOtherOne(){
+  runCoBG 1 2 3
+    runGrepSed "export BC=" "" ./"$APP" create --roster public.toml --interval .5s
+  eval $SED
+  [ -z "$BC" ] && exit 1
+
+  testOK ./"$APP" darc_add -out_key ./key.txt -out_id ./id.txt
+  KEY=`cat ./key.txt`
+  ID=`cat ./id.txt`
+  testOK ./"$APP" darc_rule_add spawn:darc -identity "$KEY" -darc "$ID" -signer "$KEY"
+  testOK ./"$APP" darc_add -darc "$ID" -sign "$KEY"
+  testOK ./"$APP" darc_show -id "$ID"
+}
+
+testAddDarcWithOwner(){
+  runCoBG 1 2 3
+    runGrepSed "export BC=" "" ./"$APP" create --roster public.toml --interval .5s
+  eval $SED
+  [ -z "$BC" ] && exit 1
+
+  testOK ./"$APP" key -save ./key.txt
+  KEY=`cat ./key.txt`
+  echo "Generated key: $KEY"
+  testOK ./"$APP" darc_add -owner "$KEY"
+}
+
+testExpression(){
+  runCoBG 1 2 3
+    runGrepSed "export BC=" "" ./"$APP" create --roster public.toml --interval .5s
+  eval $SED
+  [ -z "$BC" ] && exit 1
+
+  testOK ./"$APP" darc_add -out_id ./darc_id.txt -out_key ./darc_key.txt
+  ID=`cat ./darc_id.txt`
+  KEY=`cat ./darc_key.txt`
+  testOK ./"$APP" key -save ./key.txt
+  KEY2=`cat ./key.txt`
+
+  testOK ./"$APP" darc_rule_add spawn:darc -identity "$KEY | $KEY2" -darc "$ID" -signer "$KEY"
+  testOK ./"$APP" darc_show -id "$ID"
```

Comment:
Testing darc_show should actually grep for something, or else it could succeed without working correctly.